### DISCUSSION TITLE
foreach like pattern

### DISCUSSION THREAD

**Discussion by mipmip** [tweag/nickel#1383] [Q&A] [answered]
Say I have a array of strings with usernames. Can I iterate over this array creating a new array with records?

in array:
```
["jane" "pete" "richie"]
```

results:
```
[
  {
    username = "jane",
    email = "jane@nickel.org"
  },
  {
    username = "pete",
    email = "pete@nickel.org"
  },
  {
    username = "richie",
    email = "richie@nickel.org"
  },
]
```

**Comment by vkleen** [comment, marked as answer]
This is a textbook application for `std.array.map`:
```nickel
["jane", "pete", "richie"]
|> std.array.map
  (
    fun name =>
      {
        username = name,
        email = "%{name}@nickel.org"
      }
  )
```
In general, a foreach like pattern where the elements are handled independently from each other calls for [`std.array.map`](https://nickel-lang.org/stdlib/std-array/#map). If you need to manipulate some state depending on each element to arrive at a final value, you'd typically want to look into [`std.array.fold_left`](https://nickel-lang.org/stdlib/std-array/#fold_left).
